eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java

```diff
@@ -360,13 +364,85 @@ private List<JsonElement> getJSONAssignments(AssignmentTestCase testCase) {
         return (List<JsonElement>) this.getAssignments(testCase, AssignmentValueType.JSON);
     }
 
-    private static String getMockRandomizedAssignmentResponse() {
+    @Test
+    public void testInvalidConfigJSON() {
+
+        // Create a mock instance of EppoHttpClient
+        EppoHttpClient mockHttpClient = mock(EppoHttpClient.class);
+
+        doAnswer(invocation -> {
+            RequestCallback callback = invocation.getArgument(1);
+            callback.onSuccess(new StringReader("{}"));
+            return null; // doAnswer doesn't require a return value
+        }).when(mockHttpClient).get(anyString(), any(RequestCallback.class));
+
+        Field httpClientOverrideField = null;
+        try {
+            // Use reflection to set the httpClientOverride field
+            httpClientOverrideField = EppoClient.class.getDeclaredField("httpClientOverride");
+            httpClientOverrideField.setAccessible(true);
+            httpClientOverrideField.set(null, mockHttpClient);
+
+
+            initClient(TEST_HOST, true, true, false);
+        } catch (InterruptedException | NoSuchFieldException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (httpClientOverrideField != null) {
+                try {
+                    httpClientOverrideField.set(null, null);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException(e);
+                }
+                httpClientOverrideField.setAccessible(false);
+            }
+        }
+
+        String result = EppoClient.getInstance().getStringAssignment("dummy subject", "dummy flag");
+        assertNull(result);
+    }
+
+    @Test
+    public void testCachedBadResponseAllowsLaterFetching() {
+        // Populate the cache with a bad response
+        ConfigCacheFile cacheFile = new ConfigCacheFile(ApplicationProvider.getApplicationContext());
+        cacheFile.delete();
         try {
-            InputStream in = ApplicationProvider.getApplicationContext().getAssets()
-                    .open("rac-experiments-v3-hashed-keys.json");
-            return IOUtils.toString(in, Charsets.toCharset("UTF8"));
+            cacheFile.getOutputWriter().write("{}");
+            cacheFile.getOutputWriter().close();
         } catch (IOException e) {
-            throw new RuntimeException("Error reading mock RAC data", e);
+            throw new RuntimeException(e);
+        }
+        try {
+            initClient(TEST_HOST, false, false, false);
+        } catch (InterruptedException e) {
```

Comment:
Side note, I tried refactoring `throws InterruptedException` out of `initClient()` signature ([example commit](https://github.com/Eppo-exp/android-sdk/pull/38/commits/979018d122d1757d935f328ac469287cbfd3e7a9)), but it seemed to (inexplicably) be what was breaking `testCachedAssignments()` the most 🤷